packages/synapse-react-client/src/lib/containers/widgets/query-filter/FacetFilterControls.tsx

```diff
@@ -135,40 +137,105 @@ export const applyChangesToRangeColumn = (
   onChangeFn(result)
 }
 
-export function FacetFilterControls(props: FacetFilterControlsProps) {
-  const { facetsToFilter: facetFiltersToShow } = props
-  const { data, isLoadingNewBundle, getLastQueryRequest, executeQueryRequest } =
-    useQueryContext()
+/**
+ * Determines which facet filters should be shown after loading a new bundle. The shown facets will be the first
+ * three available facets, plus any other facets that have a filter applied.
+ * @param facets
+ * @param selectedFacets
+ * @returns the columnNames of the facets that should be shown.
+ */
+export function getDefaultShownFacetFilters(
+  facets: FacetColumnResult[],
+  selectedFacets?: FacetColumnRequest[],
+): Set<string> {
+  const columnsWithExistingFilters = (selectedFacets ?? []).map(
+    fcr => fcr.columnName,
+  )
+  return new Set([
+    ...facets.slice(0, 3).map(f => f.columnName),
+    ...columnsWithExistingFilters,
+  ])
+}
 
-  const facets = data?.facets?.filter(
-    facet =>
-      // Don't show facets where there are no values
-      !isSingleNotSetValue(facet),
+function FacetFilterSkeleton() {
+  const facetFilterFormRow = (
+    <Stack direction={'row'} gap={'10px'} sx={{ my: 1 }}>
+      <Skeleton width={'15px'} />
+      <Skeleton width={'40%'} />
+    </Stack>
   )
+  return (
+    <Box className={'FacetFilterControls__facet'}>
+      <Box className={'FacetFilterHeader'}>
+        <Skeleton width={'100%'} />
+      </Box>
+      <Box>
+        {facetFilterFormRow}
+        {facetFilterFormRow}
+        {facetFilterFormRow}
+      </Box>
+    </Box>
+  )
+}
 
-  let shownChips: string[]
-  if (facetFiltersToShow == null) {
-    shownChips = facets?.map(facet => facet.columnName) ?? []
-  } else {
-    // The facets to show have been explicitly specified
-    shownChips = facetFiltersToShow
-  }
-  const [facetFiltersShown, setFacetFiltersShown] = React.useState<string[]>([])
+function FacetFilterControlsSkeleton() {
```

Comment:
This component really needed a better loading state, so I added this simple skeleton.